back/engines/commercial/content_builder/spec/services/content_builder/layout_image_service_spec.rb

```diff
@@ -0,0 +1,232 @@
+require 'rails_helper'
+
+describe ContentBuilder::LayoutImageService do
+  let(:service) { described_class.new }
+
+  describe 'swap_data_images' do
+    it 'removes the src attribute from image elements' do
+      layout_image = create :layout_image
+      input = {
+        'ROOT' => {
+          'type' => 'div',
+          'isCanvas' => true,
+          'props' => {
+            'id' => 'e2e-content-builder-frame',
+            'style' => {
+              'padding' => '4px',
+              'minHeight' => '160px',
+              'backgroundColor' => '#fff'
+            }
+          },
+          'displayName' => 'div',
+          'custom' => {},
+          'hidden' => false,
+          'nodes' => ['XGtvXcaUr3', 'nt24xY6COf'],
+          'linkedNodes' => {}
+        },
+        'nt24xY6COf' => {
+          'type' => {
+            'resolvedName' => 'Image'
+          },
+          'isCanvas' => false,
+          'props' => {
+            'imageUrl' => layout_image.image.url,
+            'id' => 'image',
+            'alt' => '',
+            'dataCode' => layout_image.code
+          },
+          'displayName' => 'Image',
+          'custom' => {},
+          'parent' => 'ROOT',
+          'hidden' => false,
+          'nodes' => [],
+          'linkedNodes' => {}
+        },
+        'XGtvXcaUr3' => {
+          'type' => {
+            'resolvedName' => 'Text'
+          },
+          'isCanvas' => false,
+          'props' => {
+            'text' => '<p>This is some text.</p>',
+            'id' => 'text'
+          },
+          'displayName' => 'Text',
+          'custom' => {},
+          'parent' => 'ROOT',
+          'hidden' => false,
+          'nodes' => [],
+          'linkedNodes' => {}
+        }
+      }
+      imageable = build :layout, craftjs_jsonmultiloc: { 'nl-BE' => input }
+      output = service.swap_data_images imageable, :craftjs_jsonmultiloc
```

Comment:
I would move this line just above the expectation, so that the call of the method under test is close to the expectation. Then the test reads like:
```
define input
define expected result
call method under test with input
verify that the method returns the expected result
```
This applies for the other example as well.